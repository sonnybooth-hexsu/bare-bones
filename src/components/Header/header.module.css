@value mqs: "../../styles/mqs.module.css";
@value fromMedium, fromLarge from mqs;

.headerContainer {
  background-color: theme('colors.white');
}

.header {
  display: flex;
  align-items: center;
  justify-content: space-between;
  color: theme('colors.black');
  font-size: 1.8rem;
  padding: .5rem 0;
}

.headerLogo {
  font-size: 1.5rem;
  font-weight: 600;

  @media (fromLarge) {
    flex-grow: 1;
    font-size: 2rem;
  }
}

.headerNavigation {
  composes: listInline from "../../styles/helpers.module.css";
  display: none;

  @media (fromLarge) {
    display: flex;
  }
}

.headerNavigation ol {
  display: flex;
}

.headerNavigation li {
  display: inline-block;
  position: relative;
  margin-right: 2rem;
  font-size: 1rem;
  font-weight: 500;
  color: theme('colors.black');
  letter-spacing: 1;
  cursor: pointer;
}

.headerNavigation li > div {
  position: absolute;
  background-color: theme('colors.white');
  padding: 0.5rem 1.5rem 1rem;
  top: 100%;
}

.headerNavigation li > div a {
  margin-top: 0.5rem;
}

.headerNavigation li:last-of-type {
  margin-right: 0;
}

.headerNavigation li a {
  display: block;
  font-size: 1rem;
  color: theme('colors.black');
  text-decoration: none;
  font-weight: 500;
  letter-spacing: 1;
}

.headerNavigationSubMenu {
  display: flex;
  align-items: center;
}

.headerNavigationSubMenu svg {
  margin-left: 0.25rem;
  margin-top: 2px;
}

.headerMenuIcon {
  cursor: pointer;

  @media (fromLarge) {
    display: none;
  }
}
