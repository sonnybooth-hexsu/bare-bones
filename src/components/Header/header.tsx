import React from "react"
import { useStaticQuery, graphql } from "gatsby"
import styles from "./header.module.css"
import { Menu, ChevronDown } from "react-feather"
import { Search } from "../Search"
import { Whatsapp } from "../Whatsapp"

type HeaderProps = {
  siteTitle: string
  navLinks: Array<{
    name: string
    page: string
    id: number
    subLinks: Array<{ name: string; page: string; id: number }>
  }>
  navToggle: () => Function
  pageSelected: string
}

export const Header = ({
  siteTitle,
  navLinks,
  navToggle,
  pageSelected,
}: HeaderProps) => {
  const data = useStaticQuery(
    graphql`
      query {
        allFile(filter: { sourceInstanceName: { eq: "products" } }) {
          edges {
            node {
              childMarkdownRemark {
                frontmatter {
                  layout
                  title
                  category
                  path
                  excerpt
                  image
                  productAttributeTitleOne
                  productAttributeTitleTwo
                  productAttributeTitleThree
                  productAttributeTitleFour
                  productAttributeValueOne
                  productAttributeValueTwo
                  productAttributeValueThree
                  productAttributeValueFour
                }
              }
            }
          }
        }
        site {
          siteMetadata {
            telephone
          }
        }
      }
    `
  )

  return (
    <div className={styles.headerContainer}>
      <div className="container mx-auto pl-6 pr-6">
        <div className="pl-4 pr-4">
          <div className={styles.header}>
            <div className={styles.headerLogo}>
              <a href="/">{siteTitle}</a>
            </div>
            {data && <Search items={data.allFile.edges} />}
            <nav className={styles.headerNavigation}>
              <ol>
                {navLinks.map(({ name, page, subLinks, id }) => {
                  return subLinks.length > 0 ? (
                    <li tabIndex={0} key={`navigation-item-${id}`}>
                      <span className={styles.headerNavigationSubMenu}>
                        {name}
                        <ChevronDown />
                      </span>
                      <div>
                        {subLinks.map(({ name, page, id }) => (
                          <a key={`navigation-sublink-item-${id}`} href={page}>
                            {name}
                          </a>
                        ))}
                      </div>
                    </li>
                  ) : (
                    <li
                      className={`${pageSelected === name ? "underline" : ""}`}
                      data-testid="headerNavigationLink"
                      key={`navigation-item-${id}`}
                    >
                      <a href={`${page}`}>{name}</a>
                    </li>
                  )
                })}
              </ol>
            </nav>
            {data && <Whatsapp telephone={data.site.siteMetadata.telephone} />}
            <Menu onClick={navToggle} className={styles.headerMenuIcon} />
          </div>
        </div>
      </div>
    </div>
  )
}

export default Header
