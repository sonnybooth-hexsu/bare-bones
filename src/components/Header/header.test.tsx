/* eslint no-undef: 0 */

import React from "react"
import { render, cleanup } from "@testing-library/react"
import { queryAllByTestId } from "@testing-library/dom"
import Header from "./header"

const testNavLinks = [
  {
    name: `Link 1`,
    page: `link-1`,
    subLinks: [],
    id: 1,
  },
  {
    name: `Link 2`,
    page: `link-2`,
    subLinks: [],
    id: 2,
  },
  {
    name: `Link 3`,
    page: `link-3`,
    subLinks: [],
    id: 3,
  },
]

afterEach(() => {
  cleanup()
})

describe("<Header />", () => {
  it("should match snapshot", () => {
    const { container } = render(
      <Header
        siteTitle="Test Site"
        navLinks={testNavLinks}
        navToggle={() => Function}
        pageSelected="Home"
      />
    )

    expect(container).toMatchSnapshot()
  })

  it("should have the length of links passed through", () => {
    const { container } = render(
      <Header
        siteTitle="Test Site"
        navLinks={testNavLinks}
        navToggle={() => Function}
        pageSelected="Home"
      />
    )

    expect(queryAllByTestId(container, "headerNavigationLink")).toHaveLength(
      testNavLinks.length
    )
  })
})
