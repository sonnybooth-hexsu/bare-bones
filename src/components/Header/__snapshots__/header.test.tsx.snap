// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshot 1`] = `
<div>
  <div
    class="headerContainer"
  >
    <div
      class="container mx-auto pl-6 pr-6"
    >
      <div
        class="pl-4 pr-4"
      >
        <div
          class="header"
        >
          <div
            class="headerLogo"
          >
            <a
              href="/"
            >
              Test Site
            </a>
          </div>
          <nav
            class="headerNavigation"
          >
            <ol>
              <li
                data-testid="subLinkMenu"
                tabindex="0"
              >
                <span
                  class="headerNavigationSubMenu"
                >
                  Link 1
                  <svg
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <polyline
                      points="6 9 12 15 18 9"
                    />
                  </svg>
                </span>
                <div
                  class="hidden"
                  data-testid="subLinkMenuList"
                >
                  <a
                    href="Sub link-1"
                  >
                    Sub Link 1
                  </a>
                </div>
              </li>
              <li
                class=""
                data-testid="headerNavigationLink"
              >
                <a
                  href="link-2"
                >
                  Link 2
                </a>
              </li>
              <li
                class=""
                data-testid="headerNavigationLink"
              >
                <a
                  href="link-3"
                >
                  Link 3
                </a>
              </li>
            </ol>
          </nav>
          <svg
            class="headerMenuIcon"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="3"
              x2="21"
              y1="12"
              y2="12"
            />
            <line
              x1="3"
              x2="21"
              y1="6"
              y2="6"
            />
            <line
              x1="3"
              x2="21"
              y1="18"
              y2="18"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
